### DISCUSSION TITLE
Compound hover states

### DISCUSSION THREAD

**Discussion by nandorojo** [tamagui/tamagui#360] [General]
Take the following layout (which doesn't exist now, but I think should):

```tsx
<Button>
  <Button.Icon icon={lucide.Open} />
  <Button.Text>Open</Button.Text>
</Button>
```

How can I make the `Button.Text` animate based on the `Button`'s hover state? For example, I might want to change the text color when the button is hovered. Linear does this often, it looks nice.

`moti/interactions` achieves this with a combination of nested contexts. Curious what you think.

**Comment by astahmer** [comment]
Chakra-UI solves it very nicely using `_groupHover` (for your example), but also if you wanted `Button.Icon` to trigger it you could use `_peerHover`
https://chakra-ui.com/docs/styled-system/style-props#pseudo

**Comment by izakfilmalter** [comment]
@nandorojo Did you end up figuring out the animation? Ran into this yesterday and gave up.

**Reply by nandorojo** [reply]
Nope, you'd need to use onHoverIn I think.
